.github/workflows/ci.yml

```diff
@@ -0,0 +1,62 @@
+name: CI
+'on':
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+jobs:
+  # lint-dockerfile:
+  #   name: Lint Dockerfile
+  #   runs-on: ubuntu-latest
+  #   steps:
+  #     - name: Git checkout
+  #       uses: actions/checkout@v2
+  #     - name: Run Hadolint Dockerfile Linter (Common)
+  #       uses: burdzwastaken/hadolint-action@master
+  #       env:
+  #         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  #         HADOLINT_ACTION_DOCKERFILE_FOLDER: docker/cygnus-common
+  #     - name: Run Hadolint Dockerfile Linter (NGSI)
+  #       uses: burdzwastaken/hadolint-action@master
+  #       env:
+  #         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  #         HADOLINT_ACTION_DOCKERFILE_FOLDER: docker/cygnus-ngsi
+  #     - name: Run Hadolint Dockerfile Linter (NGSI-LD)
+  #       uses: burdzwastaken/hadolint-action@master
+  #       env:
+  #         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  #         HADOLINT_ACTION_DOCKERFILE_FOLDER: docker/cygnus-ngsi-ld
+  #     - name: Run Hadolint Dockerfile Linter (Twitter)
+  #       uses: burdzwastaken/hadolint-action@master
+  #       env:
+  #         GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  #         HADOLINT_ACTION_DOCKERFILE_FOLDER: docker/cygnus-twitter
+  
+  unit-test:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    steps:
```

Comment:
The idea is to "deconstruct" the test.sh script (now deleted) into several steps in ci.yml? Is my undersetanding correct?

Reply:
Yes, the return code from each runstep is the equivalent of each test

Reply:
Thanks for the clasifications. Side comments provided. NTC regarding this one.